src/core/eventloop.rs

```diff
@@ -338,8 +360,11 @@ impl<C: Callback> EventLoop<C> {
             .expect("slab should have capacity"))
     }
 
-    pub fn deregister(&self, id: usize) {
-        self.regs.remove(id);
+    pub fn deregister(&self, id: usize) -> Result<(), EventLoopError> {
+        match self.regs.remove(id) {
+            Ok(()) => Ok(()),
+            Err(_) => Err(EventLoopError),
+        }
```

Comment:
This can be simplified down since we're just throwing away the error:

```suggestion
 self.regs.remove(id).map_err(|_| EventLoopError)
```